pytorch_lightning/trainer/trainer.py

```diff
@@ -1307,7 +1307,13 @@ def _run_sanity_check(self, ref_model):
         using_val_step = self._data_connector._val_dataloader_source.is_defined() and is_overridden(
             "validation_step", ref_model
         )
-        should_sanity_check = using_val_step and self.num_sanity_val_steps > 0 and self.limit_val_batches > 0
+        should_sanity_check = (
+            using_val_step
+            and self.num_sanity_val_steps > 0
+            and self.limit_val_batches > 0
```

Comment:
can also yous `trainer.enable_validation` to avoid explicit check for `val_step` and `limit_val_batches` here.

Reply:
Done